### ISSUE TITLE
with_items failure with yum

### ISSUE THREAD

**Issue by asad-at-srt** [ansible/ansible#2258]
Variable resolution fails with yum, but succeeds with copy
Playbook looks as follows:

  vars:
     pkgs:
       - { name: "jdk", version: "1.6.0_31-fcs" }
       - { name: "httpd", version: "2.2.15-15.el6.centos.1" }

  tasks:
     - name: New style variable resolution
       action: copy src=/tmp/${item.name}-${item.version} dest=/tmp
       with_items: $pkgs

```
- name: .. but does not work with the yum module
  action: yum name=${item.name}-${item.version} state=installed
  with_items: $pkgs
```

I got the following:

PLAY [Test Playbook] ********************\* 

GATHERING FACTS ********************\* 
ok: [10.120.210.61]

TASK: [New style variable resolution] ********************\* 
ok: [10.120.210.61] => (item={'version': '1.6.0_31-fcs', 'name': 'jdk'}) => {"changed": false, "dest": "/tmp/jdk-1.6.0_31-fcs", "group": "root", "item": {"name": "jdk", "version": "1.6.0_31-fcs"}, "md5sum": "d41d8cd98f00b204e9800998ecf8427e", "mode": "0644", "owner": "root", "src": "/home/ansible/.ansible/tmp/ansible-1362168335.16-67188730965918/jdk-1.6.0_31-fcs", "state": "file"}
ok: [10.120.210.61] => (item={'version': '2.2.15-15.el6.centos.1', 'name': 'httpd'}) => {"changed": false, "dest": "/tmp/httpd-2.2.15-15.el6.centos.1", "group": "root", "item": {"name": "httpd", "version": "2.2.15-15.el6.centos.1"}, "md5sum": "d41d8cd98f00b204e9800998ecf8427e", "mode": "0644", "owner": "root", "src": "/home/ansible/.ansible/tmp/ansible-1362168335.63-42039443662886/httpd-2.2.15-15.el6.centos.1", "state": "file"}

TASK: [.. but does not work with the yum module] ********************\* 
fatal: [10.120.210.61] => Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/ansible/runner/**init**.py", line 237, in _executor
    exec_rc = self._executor_internal(host)
  File "/usr/lib/python2.6/site-packages/ansible/runner/__init__.py", line 287, in _executor_internal
    inject['item'] = ",".join(items)
TypeError: sequence item 0: expected string, dict found

FATAL: all hosts have already failed -- aborting

**Comment by mpdehaan** [comment]
Thanks for the report.  I'll make this so this optimization only occurs when it is using string data for with_items.

**Comment by mpdehaan** [comment]
Fixed in 7ac5e46
